src/snowflake/snowpark/modin/plugin/_internal/utils.py

```diff
@@ -1052,11 +1062,33 @@ def create_ordered_dataframe_from_pandas(
     # np.nan, pd.NA, pd.NaT and None all as missing value.
     isna_data = df.isna().to_numpy().tolist()
 
+    snowpark_pandas_types = []
```

Comment:
I would suggest refactoring it in a different way. The name of this method `create_ordered_dataframe_from_pandas` is about to create the ODF only and there is nothing to do with `SnowparkPandasType`. Instead, we could have another helper method called `infer_snowpark_pandas_type(df) -> tuple[snowflake_type, struct_fields]`. Then we can use this struct_fields as an input to `create_ordered_dataframe_from_pandas`.

Reply:
Good idea! Constructing each `StructField` requires the Snowflake quoted identifier, so `infer_snowpark_pandas_type` should return `DataType` but `StructField` objects. Done.